src/lib.rs

```diff
@@ -1,6 +1,6 @@
 //! This crate abstracts the network layer of a P2P network and provides a simple interface to connect to other peers.
 //! Simple example with two peers on the same code to demonstrate:
-//! ```rust
+//! ```ignore
```

Comment:
it's not possible to maintain this up to date ? So that the `lib.rs` the entrypoint of docs always have a working example

Reply:
Fixed this, I thought it was not intended to really work my bad
(Never heard of doctests before, it's nice ! :smiley: )